Headers/Public/VLCMediaPlayer.h

```diff
@@ -875,6 +889,28 @@ extern NSString *const VLCTitleDescriptionIsMenu;
 @property (NS_NONATOMIC_IOSONLY, readonly) NSImage *lastSnapshot;
 #endif
 
+/**
+ * Start recording at given **directory** path
+ * \param path directory where the recording should go
+ * \return an BOOL with the success status
+ */
+- (BOOL)startRecordingAtPath:(NSString *)path;
+
+/**
+ * Stop current recording
+ * \return an BOOL with the success status
+ */
+- (BOOL)stopRecording;
+
+/**
+ * mux srt and mp4 file to an mp4 file with embedded subtitles
+ * \param srtPath path to srt file
+ * \param mp4Path path to mp4 file
+ * \param outPath path where the new file should be written to
+ * \return an BOOL with the success status
+ */
+- (BOOL)muxSubtitleFile:(NSString *)srtPath toMp4File:(NSString *)mp4Path outputPath:(NSString *)outPath;
```

Comment:
we should add nonnull to all params

Reply:
We should move that to different PR indeed. I believe that legitimate use of nil is pretty rare throughout VLCKit indeed.